src/Symfony/Component/Ldap/Security/LdapUserProvider.php

```diff
@@ -30,6 +30,11 @@
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  * @author Charles Sarrazin <charles@sarraz.in>
  * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @template TUser of PasswordAuthenticatedUserInterface&UserInterface
```

Comment:
This is wrong. the type for this provider is `LdapUser`

Reply:
Let's not make this generic then.